includes/class-freemius.php

```diff
@@ -7544,6 +7545,45 @@ function _activate_plugin_event_hook() {
             $this->_storage->was_plugin_loaded = true;
         }
 
+        /**
+         * @author Leo Fajardo (@leorw)
+         * @since 2.8.0
+         *
+         * @param string $plugin
+         * @param bool   $network_wide
+         */
+        function _activated_plugin( $plugin, $network_wide ) {
+            // We cannot rely on fs_remove_sdk_reference_by_basename() for reordering, as the latest SDK reference might be associated with a different product, leading to the reordering of that product instead of the intended one.
+            $this->move_free_plugin_to_end_of_active_plugins( $this->_free_plugin_basename );
+
+            if ( $plugin !== $this->_free_plugin_basename ) {
+                /**
+                 * To avoid placing the free version at the start of the active plugins option when the SDK is loaded from it, remove the SDK reference linked to the free version if it is the newest.
+                 */
+                fs_remove_sdk_reference_by_basename( $this->_free_plugin_basename );
+            }
+        }
+
+        /**
+         * @author Leo Fajardo (@leorw)
+         * @since 2.8.0
+         *
+         * @param string $plugin
+         */
+        private function move_free_plugin_to_end_of_active_plugins( $plugin ) {

```

Comment:
Don't we want to handle network wide activation too?

![CleanShot 2024-08-20 at 11 35 31@2x](https://github.com/user-attachments/assets/433ecac4-c11c-4338-9393-76647b7089f5)

Maybe it will be difficult to support network wide. Since I see the `wp_get_active_network_plugins` function does a `sort` when being called. If that's the case then please add some comment here.